waveform_modes.py

```diff
@@ -315,6 +315,50 @@ def indices(self, ell_m):
             raise ValueError("Input `ell_m` should be an Nx2 sequence of integers")
         return ell_m[:, 0] * (ell_m[:, 0] + 1) - self.ell_min ** 2 + ell_m[:, 1]
 
+    def apply_eth(self, N_eth=0, N_ethbar=0, eth_convention='NP'):
+        """Apply spin raising/lowering operators to waveform mode data.
+
+        Parameters
+        ----------
+        N_eth: int
+            Number of times to apply the spin raising operator
+        N_ethbar: int
+            Number of times to apply the spin lowering operator
+        eth_convention: either 'NP' or 'GHP' 
+            Choose between Newman-Penrose or GHP convention
+                Choose between Newman-Penrose or GHP convention.
+
+        Returns
+        -------
+        mode_data: array of complex 
+
+        """
+        import spherical_functions as sf
+        if eth_convention == 'NP':
+            operator = sf.eth_NP
+            antioperator = sf.ethbar_NP
+        elif eth_convention == 'GHP':
+            operator = sf.eth_GHP
+            antioperator = sf.ethbar_GHP
+        s = self.spin_weight
```

Comment:
Add an `else` clause that raises a ValueError and explains that it has to be either NP or GHP.